### DISCUSSION TITLE
Provide a `-i` for neomutt to support git send-email?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [neomutt/neomutt#4515] [Development]
tl;dr:

Provide a `-i` for `neomutt`. Reason is <https://github.com/neomutt/neomutt/issues/3554#issuecomment-1697401830>.

<https://lore.kernel.org/git/20221106090233.ukf6rjzrygpnavzh@desktop/T/#t>

Should we provide a wrapper like `git-neomutt` to support `git send-email`? And can the tips be shared to <https://neomutt.github.io>? Thanks!

**Comment by flatcap** [comment]
I'm happy to add more contrib to NeoMutt (and share tips on the website).

What does `git-neomutt` need to do?
Where does it fit in your workflow?
Can NeoMutt already provide the functions/data you need?
If not, what's needed?

(Is there a `git-mutt` we can copy? ;-)
(sorry I don't have time to investigate myself)

**Comment by Freed-Wu** [comment]
> What does git-neomutt need to do?

git-neomutt should provide a sendmail compatible command line options for git send-email.
Such as `git send-email` want to call `sendmail -i to@mail`, now we

```dosini
[sendemail]
	suppresscc = author
	sendmailCmd = git-neomutt
```

in `~/.config/git/config`, so git call `git-neomutt -i to@email`.

`~/.local/bin/git-neomutt`:

```sh
#!/usr/bin/env bash
# https://github.com/neomutt/neomutt/issues/3554
neomutt "${@//-i/}"
```

call `neomutt to@email`. It can work.

> Where does it fit in your workflow?

```sh
git clone --depth=1 XXXX
vi XXX
git commit
git send-email -1 --to=to@email --subject='fix a bug' --cc=cc@email
```

> Can NeoMutt already provide the functions/data you need?

No.

> If not, what's needed?

Provide a sendmail compatible command line options for git send-email.

> (Is there a git-mutt we can copy? ;-)

Right.

**Comment by Freed-Wu** [comment]
Hey. A long time have passed. Let me conclude the issue:

git has a subcommand `send-email`. It will call `foobar` if your `~/.config/git/config` is:

```
[sendemail]
	sendmailCmd = foobar
```

I hope it can call `neomutt`, but I get failure:

```
neomutt: option requires an argument -- 'i'
```

<https://lore.kernel.org/git/20221106090233.ukf6rjzrygpnavzh@desktop/T/#t>
said:

> That program just has to implement a sendmail-compatible interface,
> which neomutt doesn't.

I search `sendmail` and there are not only one programs contain it. One is `msmtp`:

```sh
$ sendmail --help
Usage:
...
Accepted but ignored: -A, -B, -bm, -F, -G, -h, -i, -L, -m, -n, -O, -o, -v
```

Another is `busybox`:

```sh
$ busybox sendmail --help
...
Standard options:
...
        -i              -oi synonym, implied and ignored
```

So We have two methods:

1. let `neomutt` support `-i`, but ignore it like `sendmail`.
2. provide a wrapper, such as:

`git-neomutt`:

```sh
#!/usr/bin/env sh
neomutt ${@//-i/}
```

Then let user use it.

Which is better? Any discussion is welcome.

**Comment by ossilator** [comment]
the wrapper is the obvious way to go.
but i don't think it belongs into mutt, as a sendmail command line adapter  really isn't a natural thing for a MUA.
in fact, i don't understand why you want this at all. what value does this add over an existing solution?

**Comment by Freed-Wu** [comment]
> what value does this add over an existing solution

Keep compatibility with `sendmail`, which make it can support `git send-email`. Do there any other one who use `git send-email`?

**Comment by ossilator** [comment]
> Keep compatibility with sendmail, which make it can support git send-email.

this doesn't answer the question. why mutt instead of nullmailer, msmtp, whatever?

**Comment by Freed-Wu** [comment]
> why mutt instead of nullmailer, msmtp, whatever?

I don't want to install too many similar programs. If mutt can do, use mutt.

**Comment by ossilator** [comment]
a MUA and an MTA are not exactly similar programs. mutt now incorporates an SMTP client like every modern MUA does, but (ab-)using it as a nullmailer seems a bit weird to me.

**Comment by Freed-Wu** [comment]
> not exactly similar programs

Sorry for my not exact words, I means mutt can do the work of sendmail, so from this point, they are similar.
